### PR TITLE
Remove testbed and replace sign_in() and sign_out()

### PR DESCRIPTION
**Pull request by KyleJu**
Part one of #1077

Hacked the setup by adding `os.environ['AUTH_DOMAIN'] = '1'`. Without this line, any tests related to `from google.appengine.api import users as gae_users` will fail.

### CODE REVIEW

File: testing_config.py [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
 app_engine_path = os.environ.get('APP_ENGINE_PATH', '')
 if not app_engine_path:
-  app_engine_path = '/usr/lib/google-cloud-sdk/platform/google_appengine'
+  app_engine_path = '/usr/local/google/home/kyleju/google-cloud-sdk/platform/google_appengine'
```
Comment (KyleJu) [inline testing_config.py:25]:
```suggestion
  app_engine_path = '/usr/lib/google-cloud-sdk/platform/google_appengine'
```
Revert this change

**Comment by KyleJu** [comment]
@jrobbins After some trial and error, I removed `testbed` without using mock. I also tried an alternative approach that requires mocking `from google.appengine.api import users as gae_users`. IIUC, we have to remove `google.appengine.api ` (introduced in #1310) for py3. I do see the comments `# TODO(jrobbins): phase out gae_users.` Any update on this?

**Comment by KyleJu** [comment]
CI is failing but the `npm test` passed on my local machine:

```
EEEEEEEEEEEEEEEEEETraceback (most recent call last):
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/unittest/__main__.py", line 12, in <module>
    main(module=None)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/unittest/main.py", line 95, in __init__
    self.runTests()
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/unittest/main.py", line 232, in runTests
    self.result = testRunner.run(self.test)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/unittest/runner.py", line 151, in run
    test(result)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/unittest/suite.py", line 70, in __call__
    return self.run(*args, **kwds)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/unittest/suite.py", line 108, in run
    test(result)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/unittest/suite.py", line 70, in __call__
    return self.run(*args, **kwds)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/unittest/suite.py", line 108, in run
    test(result)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/unittest/suite.py", line 70, in __call__
    return self.run(*args, **kwds)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/unittest/suite.py", line 108, in run
    test(result)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/unittest/case.py", line 393, in __call__
    return self.run(*args, **kwds)
  File "/home/runner/work/chromium-dashboard/chromium-dashboard/testing_config.py", line 107, in run
    client = ndb.Client()
  File "/home/runner/work/chromium-dashboard/chromium-dashboard/lib/google/cloud/ndb/client.py", line 110, in __init__
    _http=requests.Session,
  File "/home/runner/work/chromium-dashboard/chromium-dashboard/lib/google/cloud/client.py", line 318, in __init__
    _ClientProjectMixin.__init__(self, project=project, credentials=credentials)
  File "/home/runner/work/chromium-dashboard/chromium-dashboard/lib/google/cloud/client.py", line 266, in __init__
    project = self._determine_default(project)
  File "/home/runner/work/chromium-dashboard/chromium-dashboard/lib/google/cloud/ndb/client.py", line 212, in _determine_default
    return _determine_default_project(project)
  File "/home/runner/work/chromium-dashboard/chromium-dashboard/lib/google/cloud/ndb/client.py", line 67, in _determine_default_project
    project = _helpers._determine_default_project(project=project)
  File "/home/runner/work/chromium-dashboard/chromium-dashboard/lib/google/cloud/_helpers.py", line 186, in _determine_default_project
    _, project = google.auth.default()
  File "/home/runner/work/chromium-dashboard/chromium-dashboard/lib/google/auth/_default.py", line 340, in default
    credentials, project_id = checker()
  File "/home/runner/work/chromium-dashboard/chromium-dashboard/lib/google/auth/_default.py", line 208, in _get_gae_credentials
    project_id = app_engine.get_project_id()
  File "/home/runner/work/chromium-dashboard/chromium-dashboard/lib/google/auth/app_engine.py", line 77, in get_project_id
    return app_identity.get_application_id()
  File "/usr/lib/google-cloud-sdk/platform/google_appengine/google/appengine/api/app_identity/app_identity.py", line 453, in get_application_id
    _, domain_name, display_app_id = _ParseFullAppId(full_app_id)
  File "/usr/lib/google-cloud-sdk/platform/google_appengine/google/appengine/api/app_identity/app_identity.py", line 434, in _ParseFullAppId
    psep = app_id.find(_PARTITION_SEPARATOR)
AttributeError: 'NoneType' object has no attribute 'find'
```

**Comment by KyleJu** [comment]
> CI is failing but the `npm test` passed on my local machine:
> 
> ```
> EEEEEEEEEEEEEEEEEETraceback (most recent call last):
>   File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/runpy.py", line 174, in _run_module_as_main
>     "__main__", fname, loader, pkg_name)
>   File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/runpy.py", line 72, in _run_code
>     exec code in run_globals
>   File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/unittest/__main__.py", line 12, in <module>
>     main(module=None)
>   File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/unittest/main.py", line 95, in __init__
>     self.runTests()
>   File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/unittest/main.py", line 232, in runTests
>     self.result = testRunner.run(self.test)
>   File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/unittest/runner.py", line 151, in run
>     test(result)
>   File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/unittest/suite.py", line 70, in __call__
>     return self.run(*args, **kwds)
>   File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/unittest/suite.py", line 108, in run
>     test(result)
>   File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/unittest/suite.py", line 70, in __call__
>     return self.run(*args, **kwds)
>   File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/unittest/suite.py", line 108, in run
>     test(result)
>   File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/unittest/suite.py", line 70, in __call__
>     return self.run(*args, **kwds)
>   File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/unittest/suite.py", line 108, in run
>     test(result)
>   File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/unittest/case.py", line 393, in __call__
>     return self.run(*args, **kwds)
>   File "/home/runner/work/chromium-dashboard/chromium-dashboard/testing_config.py", line 107, in run
>     client = ndb.Client()
>   File "/home/runner/work/chromium-dashboard/chromium-dashboard/lib/google/cloud/ndb/client.py", line 110, in __init__
>     _http=requests.Session,
>   File "/home/runner/work/chromium-dashboard/chromium-dashboard/lib/google/cloud/client.py", line 318, in __init__
>     _ClientProjectMixin.__init__(self, project=project, credentials=credentials)
>   File "/home/runner/work/chromium-dashboard/chromium-dashboard/lib/google/cloud/client.py", line 266, in __init__
>     project = self._determine_default(project)
>   File "/home/runner/work/chromium-dashboard/chromium-dashboard/lib/google/cloud/ndb/client.py", line 212, in _determine_default
>     return _determine_default_project(project)
>   File "/home/runner/work/chromium-dashboard/chromium-dashboard/lib/google/cloud/ndb/client.py", line 67, in _determine_default_project
>     project = _helpers._determine_default_project(project=project)
>   File "/home/runner/work/chromium-dashboard/chromium-dashboard/lib/google/cloud/_helpers.py", line 186, in _determine_default_project
>     _, project = google.auth.default()
>   File "/home/runner/work/chromium-dashboard/chromium-dashboard/lib/google/auth/_default.py", line 340, in default
>     credentials, project_id = checker()
>   File "/home/runner/work/chromium-dashboard/chromium-dashboard/lib/google/auth/_default.py", line 208, in _get_gae_credentials
>     project_id = app_engine.get_project_id()
>   File "/home/runner/work/chromium-dashboard/chromium-dashboard/lib/google/auth/app_engine.py", line 77, in get_project_id
>     return app_identity.get_application_id()
>   File "/usr/lib/google-cloud-sdk/platform/google_appengine/google/appengine/api/app_identity/app_identity.py", line 453, in get_application_id
>     _, domain_name, display_app_id = _ParseFullAppId(full_app_id)
>   File "/usr/lib/google-cloud-sdk/platform/google_appengine/google/appengine/api/app_identity/app_identity.py", line 434, in _ParseFullAppId
>     psep = app_id.find(_PARTITION_SEPARATOR)
> AttributeError: 'NoneType' object has no attribute 'find'
> ```

After some quick search, looks like `os.environ['APPLICATION_ID']` needs to be set for the emulator as well.

**Review by jrobbins — APPROVED** [review]

**Comment by jrobbins** [comment]
This change looks good.  I am glad that it turned out to be straightforward.
You could set `os.environ['APPLICATION_ID']` in `testing_config.py`.

**Comment by jrobbins** [comment]
>  I do see the comments `# TODO(jrobbins): phase out gae_users.` Any update on this?

I think that today I can send you a PR to eliminate 2 of 3 references to gae_users.

The reference in models.py `isinstance(value, gae_users.User)`might be more tricky. We should first check to see if that line of code is ever reached in production any more.  So, I will add logging.

**Comment by KyleJu** [comment]
> > I do see the comments `# TODO(jrobbins): phase out gae_users.` Any update on this?
> 
> I think that today I can send you a PR to eliminate 2 of 3 references to gae_users.
> 
> The reference in models.py `isinstance(value, gae_users.User)`might be more tricky. We should first check to see if that line of code is ever reached in production any more. So, I will add logging.

Got the CI to pass and merged  :). I had a hard time bypassing `isinstance(value, gae_users.User)` in the mock as well, which prompted me to use the approach in this PR and luckily it worked
